### ISSUE TITLE
RNN Using TensorFlow 2.0

### ISSUE THREAD

**Issue by ashishsme14** [tensorflow/tensorflow#45628]
Please let us know anything wrong in below code, not getting desire result -

```
from numpy import sqrt
from numpy import asarray
from pandas import read_csv
from tensorflow.keras import Sequential
from tensorflow.keras.layers import Dense
from tensorflow.keras.layers import LSTM
import tensorflow as tf
from sklearn import metrics
from sklearn.model_selection import train_test_split
```

Assign the value as 40 to the variabel RANDOM_SEED which will be the seed value.
Set the random seed value using the value stored in the variable RANDOM_SEED.

```
RANDOM_SEED = 40
tf.random.set_seed(RANDOM_SEED)
```

split a univariate sequence into samples

```
def split_sequence(sequence, n_steps):
    X, y = list(), list()
    for i in range(len(sequence)):
        # find the end of this pattern
        end_ix = i + n_steps
        # check if we are beyond the sequence
        if end_ix > len(sequence)-1:
            break
        # gather input and output parts of the pattern
        seq_x, seq_y = sequence[i:end_ix], sequence[end_ix]
        X.append(seq_x)
        y.append(seq_y)
    return asarray(X), asarray(y)
```

Read the dataset airline-passengers.csv and give parameter index_col as 0 and save it in variable df.

`df = read_csv("airline-passengers.csv", index_col=0)
`

Convert the data type of the values dataframe df to float32 and save it in variable values.
Assign the value 5 to the variable n_steps which is the window size.
Split the samples using the function split_sequence and pass the parameters values and n_steps and save it in variables X and y

```
values = df.values.astype('float32')
n_steps = 5
X, y = split_sequence(values, n_steps)
```

Split the data X,y with the train_test_split function of sklearn with parameters test_size=0.33 and random_state=RANDOM_SEED.**

`X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=RANDOM_SEED)
`

Construct a fully-connected network structure defined using dense class
Create a sequential model
Add a LSTM layer which has 200 nodes with activation function as relu and input shape as (n_steps,1).
The first hidden layer has 100 nodes and uses the relu activation function.
The second hidden layer has 50 nodes and uses the relu activation function.
The output layer has 1 node.

```
model = Sequential()
model.add(LSTM(200, activation='relu',  input_shape=(n_steps,1)))
model.add(Dense(100, activation='relu'))
model.add(Dense(
50, activation='relu'))
model.add(Dense(1))
```

While comipling the model pass the following parameters -
-optimizer as Adam
-loss as mse
-metrics as mae

`model.compile(optimizer='Adam', loss='mse', metrics=['mae'])`

fit the model with X_train, y_train, epochs=350, batch_size=32,verbose=0.

`model.fit(X_train, y_train, epochs=350, batch_size=32, verbose=0)
`

Perform prediction on the test data (i.e) on X_test and save the predictions in the variable y_pred.

`y_pred = model.predict( X_test)`

Calculate the mean squared error on the variables y_test and y_pred using the mean_squared_error function in sklearn metrics and save it in variable MSE.
Calculate the Root mean squared error on the variables y_test and y_pred by performing square root on the above result and save it in variable RMSE.
Calculate the mean absolute error on the variables y_test and y_pred using the mean_absolute_error function in sklearn metrics and save it in variable MAE.

```
MSE  = metrics.mean_squared_error(y_test,y_pred)
RMSE = sqrt(metrics.mean_squared_error(y_test,y_pred))
MAE  = metrics.mean_absolute_error(y_test,y_pred)
print('MSE: %.3f, RMSE: %.3f, MAE: %.3f' % (MSE, RMSE,MAE))

```

**MSE: 665.522, RMSE: 25.798, MAE: 17.127 ... this we getting and it is wrong.**

```
with open("MSE.txt", "w") as text_file:
        MSE=str(MSE)
        text_file.write(MSE)
with open("RMSE.txt", "w") as text_file:
        RMSE=str(RMSE)
        text_file.write(RMSE)
with open("MAE.txt", "w") as text_file:
        MAE=str(MAE)
        text_file.write(MAE)
# serialize model to JSON
model_json = model.to_json()
with open("model.json", "w") as json_file:
    json_file.write(model_json)
```

[airline-passengers.zip](https://github.com/tensorflow/tensorflow/files/5650585/airline-passengers.zip)
[RNN_Question.zip](https://github.com/tensorflow/tensorflow/files/5650599/RNN_Question.zip)

After running code then we execute

```
from hashlib import md5
f = open("MSE.txt", "r")
s=f.read()
s=float(s)
s=round(s,3)
f1=open("RMSE.txt","r")
s1=f1.read()
s1=float(s1)
s1=round(s1,3)
f2=open("MAE.txt","r")
s2=f2.read()
s2=float(s2)
s2=round(s2,3)
if (md5(str(s).encode()).hexdigest() == '51ad543f7ac467cb8b518f1a04cc06af') and (md5(str(s1).encode()).hexdigest() == '6ad48a76bec847ede2ad2c328978bcfa') and (md5(str(s2).encode()).hexdigest() == '64bd1e146726e9f8622756173ab27831'):

	print("Your MSE,RMSE and MAE Scores matched the expected output")
else :
	print("Your MSE,RMSE and MAE Scores does not match the expected output") 
```

Here our output should be match but coming as unmatched.

**Comment by ashishsme14** [comment]
please help us

**Comment by amahendrakar** [comment]
@ashishsme14,
Looks like this is a duplicate of [#45411](https://github.com/tensorflow/tensorflow/issues/45411#issuecomment-742214832). 

And as mentioned, this question is better asked on [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow) since it is not a TensorFlow bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by ashishsme14** [comment]
No help from that

**Comment by jagadeeshjr5** [comment]
Hey, had you solved this problem. I'm also getting the same issue

**Comment by ashishsme14** [comment]
@jagadeeshjr5  did u followed same steps and getting same outut?

**Comment by jagadeeshjr5** [comment]
@ashishsme14 yes I followed my steps and got the same output

**Comment by ashishsme14** [comment]
@jagadeeshjr5  can u please tell ur steps n output please

**Comment by jagadeeshjr5** [comment]
@ashishsme14 I too got the same output, there's may be wrong in testcode or in the question

**Comment by ashishsme14** [comment]
@jagadeeshjr5  so u passed test case?

**Comment by jagadeeshjr5** [comment]
@ashishsme14 for RNN only I'm not able to pass the testcase

**Comment by ashishsme14** [comment]
please help us to review steps which we followed

**Comment by jagadeeshjr5** [comment]
@ashishsme14 raise a feedback in fresco regrading the issue

**Comment by ashishsme14** [comment]
@jagadeeshjr5  can u try now n see any luck for me no

**Comment by jagadeeshjr5** [comment]
@ashishsme14 I already did, you also raise a feedback once

**Comment by jagadeeshjr5** [comment]
@ashishsme14 there's mistake from their end

**Comment by ashishsme14** [comment]
@jagadeeshjr5  they saying they corrected it but no luck for me... that why any luck for u?

**Comment by jagadeeshjr5** [comment]
@ashishsme14 when did they told you

**Comment by ashishsme14** [comment]
@jagadeeshjr5 today

**Comment by jagadeeshjr5** [comment]
@ashishsme14 try again, if not getting passed tell them there's an issue with the question

**Comment by jagadeeshjr5** [comment]
@ashishsme14 still not able to pass

**Comment by ashishsme14** [comment]
@jagadeeshjr5 reset and install n follow rest process

**Comment by jagadeeshjr5** [comment]
@ashishsme14 have you completed

**Comment by jagadeeshjr5** [comment]
@ashishsme14 can you give me your number I'll text you, after that you can delete it

**Comment by jagadeeshjr5** [comment]
@ashishsme14 Still not able to pass

**Comment by jagadeeshjr5** [comment]
@ashishsme14 can you tell me what output you got (mse, rmse, Mar)

**Comment by jagadeeshjr5** [comment]
@ashishsme14 Can you help me with this

**Comment by ashishsme14** [comment]
@jagadeeshjr5  Once launch do not install anything. 1st re start then install n follow rest process

**Comment by jagadeeshjr5** [comment]
@ashishsme14 I did it bro, but still no luck, can you tell me what output you got(mse, rmse, mae)

**Comment by ashishsme14** [comment]
@jagadeeshjr5 not rem it ..

**Comment by jagadeeshjr5** [comment]
@ashishsme14 did you get the same as before or different

**Comment by ashishsme14** [comment]
@jagadeeshjr5 different

**Comment by jagadeeshjr5** [comment]
@ashishsme14 can you tell me the exact procedure what you did

**Comment by ashishsme14** [comment]
same steps as mentioned here
nothing different

**Comment by jagadeeshjr5** [comment]
@ashishsme14 I mean exactly what after clicking on start test

**Comment by ashishsme14** [comment]
yes same steps  RESET -> Install ? Run -> Test

**Comment by jagadeeshjr5** [comment]
@ashishsme14 on getting into the environment Reset -> Install -> Run -> Open preview -> write code -> run -> test is this the process

**Comment by ashishsme14** [comment]
yes

**Comment by jagadeeshjr5** [comment]
@ashishsme14 Still getting the same output bro, not able to pass

**Comment by jagadeeshjr5** [comment]
@ashishsme14 If you don't mind, can you reattempt it and tell me the output you got bro, that'll be very helpful

**Comment by RV97054** [comment]
Hi, is this issue was resolved successful? 
What was the solution? can you share please?
I has the same problem

**Comment by ashissahu** [comment]
MSE: 1544.323, RMSE: 39.298, MAE: 37.612

I am getting these forMLP regression.This is njot matching . Can someone help me with the correct ones ?
